kernel/common/enumerable.rb

```diff
@@ -408,11 +419,19 @@ def drop_while
   end
 
   def each_cons(num)
-    return to_enum(:each_cons, num) unless block_given?
-
     n = Rubinius::Type.coerce_to_collection_index num
     raise ArgumentError, "invalid size: #{n}" if n <= 0
 
+    return to_enum(:each_cons, num) {
+      if (enum_size = enumerator_size).nil? 
```

Comment:
I think the reason at the time was to 'cache' the value and not having to call the [enumerator_size](https://github.com/rubinius/rubinius/blob/master/kernel/common/enumerable.rb#L117-L119) method more times than actually needed.
I didn't measure it, but probably the impact is negligible vs the code readability.
What do you think?